flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamOperatorFactory.java

```diff
@@ -32,14 +30,13 @@
  *
  * @param <OUT> The output type of the operator
  */
-@Internal
+@PublicEvolving
 public interface StreamOperatorFactory<OUT> extends Serializable {
 
 	/**
 	 * Create the operator. Sets access to the context and the output.
 	 */
-	<T extends StreamOperator<OUT>> T createStreamOperator(
-			StreamTask<?, ?> containingTask, StreamConfig config, Output<StreamRecord<OUT>> output);
+	<T extends StreamOperator<OUT>> T createStreamOperator(StreamOperatorInitializer<OUT> initializer);
```

Comment:
Why do we allow client to specify return type (`T`)? 
IMO, it's a factory who knows what it creates, except for strange cases of `SimpleOperatorFactory` and `CodeGenOperatorFactory` :)

I see two options:
1. parameterize `StreamOperatorFactory` with the return type (I tried - too many changes)
1. return `StreamOperator<OUT>` and move cast to the client; this is less casts and IMO confusion